core/trino-main/src/main/java/io/trino/execution/scheduler/SqlQueryScheduler.java

```diff
@@ -615,7 +616,11 @@ private static TableInfo getTableInfo(Session session, Metadata metadata, TableS
         {
             TableSchema tableSchema = metadata.getTableSchema(session, node.getTable());
             TableProperties tableProperties = metadata.getTableProperties(session, node.getTable());
-            return new TableInfo(tableSchema.getQualifiedName(), tableProperties.getPredicate());
+            Optional<String> connectorName = metadata.listCatalogs(session).stream()
+                    .filter(catalogInfo -> catalogInfo.getCatalogName().equals(tableSchema.getCatalogName()))
+                    .map(CatalogInfo::getConnectorName)
+                    .findFirst();
```

Comment:
this should always exists, right?

Reply:
> In future if catalogs can be removed on the fly, I'm not sure if that assumption would continue to hold.

I don't think they can be removed for active query

Reply:
> this should always exists, right?

I tried making it not optional, but there were many tests which started failing. IIRC these errors were coming in the case of an information schema.